site/guides/tenant-onboarding.md

```diff
@@ -0,0 +1,274 @@
+        ├── quota.yaml
+        ├── role-binding.yaml
+        └── service-account.yaml
+```
+
+## Tenant package
+
+The tenant package should have a good set of defaults configuration so that it
+can work as a good starting point for most of the tenants. Application developer
+teams can customize the tenant package over time as their need evolves.
+
+The tenant package should include invariants (guardrails) that prevent
+mis-configuration. For example, each tenant package shouldn’t have more than
+one namespace, resource quota shouldn’t exceed limits etc.
+
+One of the key principles to keep in mind is that the tenant package shouldn’t
+try to offer all the possible customization options. The tenant package should
+offer a reasonable set of defaults with required constraints. Downstream users
+of the tenant package can directly edit, add/delete resources as per their needs.
+
+Here is an example of a [basic tenant package](https://github.com/GoogleContainerTools/kpt/tree/main/package-examples/tenant).
+
+```shell
+
+pkg-catalog $ kpt pkg tree tenant
+Package "tenant"
+├── [Kptfile]  Kptfile tenant
+├── [namespace.yaml]  Namespace tenant-name
+├── [ns-invariant.yaml]  StarlarkRun ns-invariant
+├── [quota.yaml]  ResourceQuota tenant-name/quota
+├── [role-binding.yaml]  RoleBinding tenant-name/sa-admin
+└── [service-account.yaml]  ServiceAccount tenant-name/sa
+
+```
+
+The tenant package’s Kptfile offers basic customization and enforces single
+namespace constraint.
+
+```shell
+pkg-catalog $ cat tenant/Kptfile
+apiVersion: kpt.dev/v1
+kind: Kptfile
+metadata:
+  name: tenant
+info:
+  description: Base tenant package
+pipeline:
+  mutators:
+    - image: set-namespace:v0.1
+      configMap:
+        namespace: tenant-name # ←- will be customized for pkg variant
+  validators:
+    - image: gcr.io/kpt-fn/starlark:v0.3
+      configPath: ns-invariant.yaml
+
+```
+
+### Optional resources
+
+Sometimes customizing a package may require adding a new resource. For example,
+if a team wants to define rules for isolating the network for a tenant, they
+need to define network policy resources. Our tenant package has a hidden
+directory called `.snippets` that contains snippets of such resources that a
+package consumer may need. Platform team can also surface canned configuration
+for example quota.medium.yaml can be used for medium size workloads.
+
+Also `.krmignore` files instructs kpt to ignore all resources in the `.snippets`
+directory for other workflows.
+
+```shell
+pkg-catalog $ ls tenant/.snippets/
+isolate-network-policy.yaml  quota.large.yaml  quota.medium.yaml
+quota.small.yaml  role-binding.yaml  service-account.yaml
+
+pkg-catalog $ cat tenant/.krmignore
+.snippets
+
+```
+
+### Publishing Tenant Package
+
+So once you are happy with the tenant package, you can publish the tenant
+package by tagging the version as shown below:
+
+```shell
+# Assuming you are in the pkg catalog repo where tenant package exists
```

Comment:
I think we should have e2e steps where a user can copy-paste them to achieve the task. Steps like this might stump some users. Please mention simple steps to create package-repo.